crates/turbopack/src/module_options/mod.rs

```diff
@@ -462,13 +462,16 @@ impl ModuleOptionsVc {
             for (glob, rule) in webpack_loaders_options.rules.await?.iter() {
                 rules.push(ModuleRule::new(
                     ModuleRuleCondition::All(vec![
-                        if glob.contains("/") {
+                        if glob.starts_with("./") || glob.starts_with("../") {
                             ModuleRuleCondition::ResourcePathGlob {
                                 base: execution_context.project_path().await?,
                                 glob: GlobVc::new(glob).await?,
                             }
                         } else {
-                            ModuleRuleCondition::ResourceBasePathGlob(GlobVc::new(glob).await?)
+                            ModuleRuleCondition::ResourcePathGlob {
+                                base: execution_context.project_path().await?,
+                                glob: GlobVc::new(&format!("**/{glob}")).await?,
+                            }
                         },
```

Comment:
Sorry, I think this is still incorrect. The examples being:

1. `./foo/bar.png`
2. `foo/bar.png`
3. `**/foo/bar.png`

1\. is definitely rooted in the current directory. 3\. can match any `foo/bar.png` in any directory. Currently 2\. matches 3\.'s behavior, but I think it should match 1\.'s behavior.

So we'd need 3 branches:

```rust
if glob.starts_with("./") || glob.starts_with("../") {
  // rooted
} else if glob.contains("/") {
  // prepend `./` to make it rooted
} else {
  // prepend `**/` to make it matcy any
}
```

Reply:
yeah, you are right. I also want to handle starting with `**/` to not prepend `./` in that case